technical-reports/format/composite-types.md

```diff
@@ -0,0 +1,220 @@
+  - You can think of a composite type as something "inside" an individual token.
+
+## Type checking
+
+Just as with “normal” types for tokens, using a custom, composite type will allow tools to check that the values you use match the expected type. In our color pair example above, attempting to do the following would be invalid and tools should ignore the token and show an error to the user, since “Comic Sans MS” is not a valid color value.
+
+<aside class="example" title="Invalid type">
+
+```json
+{
+  "broken-token": {
+    "type": "{my types.color pair}",
+    "value": {
+      "foreground": "Comic Sans MS",
+      "background": "#eeeeee"
+    }
+  }
+}
+```
+
+</aside>
+
+Likewise, IDEs can offer appropriate auto-completions to users that manually author design token files.
+
+## Tool support
+
+Since composite types are user-defined, tools cannot provide specialized handling of them since they have no advanced knowledge of the type declarations. Here are some suggested strategies to get the most value out of user-defined composite types:
+
+### Fallbacks
+
+Given that composite types are ultimately composed of the core design token types, tools that encounter unfamiliar composite tokens could fall back to treating their individual values as separate tokens. In effect, the composite token would be treated the same way as a group of tokens.
+
+Using the color pair type as an example, an export tool like Style Dictionary could just export 2 color variables for it. A token file like this…
+
+<aside class="example" title="JSON source">
+
+```json
+{
+  "My token": {
+    "type": "{my types.color pair}",
+    "value": {
+      "foreground": "#333333",
+      "background": "#ffffff"
+    }
+  }
+}
+```
+
+</aside>
+
+….might be exported to Sass like this:
+
+<aside class="example" title="Sass output with fallback">
+
+```css
+$my-token-foreground: #333333;
+$my-token-background: #ffffff;
+```
+
+</aside>
+
+In a similar vein, a GUI tool can "pluck" out the individual values of a composite token and use them as it would normally.
+
+E.g. a design tool like Figma might not have the concept of a color pair, so it can't do anything special with tokens of that type. However, that doesn't prevent it from displaying the foreground and background colors of those tokens alongside any plain color tokens in a color picker.
+
+### Appending custom type definitions
+
+While a tool that imports a token file containing user-defined type definitions that it doesn't understand might not be able to do anything special with them, there's nothing stopping tools that export or modify token files from adding their own type definitions to the file alongside tokens of that type.
```

Comment:
This is a very long sentence. Can we break it up for better readability?

Reply:
I tried updating it, how does it sound now?

> A tool might not understand certain user-defined type definitions in a token file and therefore can't do anything special with them. However, that tool could export or modify token files by adding their own type definitions.

Reply:
This can be re-edited later, but I feel like there's an opportunity for clear technical guidance here.

What exactly should happen when a tool doesn't understand user-defined types? Is there a cause and effect between this and the second sentence in the paragraph, or can they be two separate independent statements?

Reply:
In hindsight, this "Appending custom type definitions" section might be redundant. As it stands, composite tokens require a custom type definition. Therefore, any tool wanting to output composite tokens will necessarily need to embed the corresponding type definition in the token file.

Reply:
yeah my review feedback is more on the grammar of a long run-on sentence — I'll let y'all figure out if you want to add or remove this. :)

Reply:
created issues for the other parts of this review. this is the last item standing before I can close the review. Do we know how we want to go forward with this?